### ISSUE TITLE
Memory leakage

### ISSUE THREAD

**Issue by dimrozakis** [prometheus/blackbox_exporter#239]
I have a server running docker, and use docker-compose to deploy a monitoring stack including prometheus and blackbox exporter. Two days ago I pulled newer images for all components, including blackbox-exporter. It's been leaking memory ever since.

![blackbox-exporter-memory](https://user-images.githubusercontent.com/5428237/31054598-5b302268-a6be-11e7-8fba-345fe43e0293.png)

This is a stacked memory graph for all containers, including `blackbox_exporter`. Data taken by cAdvisor. Notice how blackbox-exporter's memory increases linearly after the upgrade. The reset and subsequent rise was caused by a restart of the blackbox-exporter container.

### Environment info

```
$ uname -a
Linux XXXXXX 4.9.7-x86_64-linode80 #2 SMP Thu Feb 2 15:43:55 EST 2017 x86_64 GNU/Linux


$ docker version
Client:
 Version:      17.06.2-ce
 API version:  1.30
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 20:00:28 2017
 OS/Arch:      linux/amd64

Server:
 Version:      17.06.2-ce
 API version:  1.30 (minimum version 1.12)
 Go version:   go1.8.3
 Git commit:   cec0b72
 Built:        Tue Sep  5 19:59:22 2017
 OS/Arch:      linux/amd64
 Experimental: false


$ docker-compose version
docker-compose version 1.12.0, build b31ff33
docker-py version: 2.2.1
CPython version: 2.7.9
OpenSSL version: OpenSSL 1.0.1t  3 May 2016


$ docker images|grep blackbox
prom/blackbox-exporter                   latest              bc728865b433        9 days ago          16.7MB
prom/blackbox-exporter                   <none>              e3217c57425e        5 months ago        14.3MB
prom/blackbox-exporter                   <none>              f3fa0e147fb5        8 months ago        14.4MB
```

### blackbox_exporter version: output of `blackbox_exporter -version`

It's now running `prom/blackbox-exporter:v0.9.1` which suffers from memory leakage. Before that, it was running `prom/blackbox-exporter:v0.5.0` which didn't leak memory.

### What is the blackbox.yml module config.

No custom config file, whatever default configuration is stored in the official blackbox exporter docker image.

### What is the prometheus.yml scrape config.

```
  # Collect external probe metrics from blackbox-exporter
  - job_name: blackbox
    metrics_path: /probe
    params:
      module: [http_2xx]  # Look for a HTTP 200 response
    static_configs:
      - targets:  # Targets to probe
        - some-target
        - some-other-target
    relabel_configs:

      # Config required for blackbox exporter to work
      - source_labels: [__address__]
        regex: (.*)(:80)?
        target_label: __param_target
        replacement: $1
      - source_labels: [__param_target]
        regex: (.*)
        target_label: instance
        replacement: $1
      - source_labels: []
        regex: .*
        target_label: __address__
        replacement: blackbox:9115  # Blackbox exporter
```

**Comment by dimrozakis** [comment]
Is this a duplicate of #236? If so, it seems like it was fixed in #238. When can we expect a new release? 

Until then, what is the most recent blackbox-exporter version that doesn't leak memory?

**Comment by brian-brazil** [comment]
Yes it's a dupe. 0.8.1 is good.
